crates/mem-pool/src/pool.rs

```diff
@@ -1275,6 +1311,45 @@ impl MemPool {
         let tx_receipt =
             TxReceipt::build_receipt(tx.witness_hash().into(), run_result, merkle_state);
 
+        // fan-out to readonly mem block
+        if let Some(handler) = &self.fan_out_mem_block_handler {
+            handler.new_tx(tx)
+        }
+
         Ok(tx_receipt)
     }
+
+    // Only **ReadOnly** node needs this.
+    // Refresh mem block with those params.
+    pub(crate) fn refresh_mem_block(
+        &mut self,
+        block_info: BlockInfo,
+        withdrawals: Vec<WithdrawalRequest>,
+        deposits: Vec<DepositInfo>,
+        txs: Vec<L2Transaction>,
+    ) -> Result<()> {
+        let mem_block = MemBlock::new(block_info, self.mem_block.prev_merkle_state().to_owned());
```

Comment:
The second block should be the local `tip_block.post_merkle_state`.

Reply:
And we should ignore this update if the block_info isn't match the local block. hint: The local tip block should be parent block of `block_info`